### PR TITLE
Remove wait package

### PR DESCRIPTION
**Pull request by mortent**
This PR does four things, each in a separate commit:
 - Remove the dependencies from the `polling` package on the `wait` package. This was only the `ResourceIdentifier` type. It instead uses the `ObjMetadata` type from the `prune` package.
 - Move the `ObjMetadata` type (with functions and tests) out of the `prune` package and into a new `object` package that it is top level under `pkg`.
 - Refactor the `Applier` to use the `polling` package rather than the `wait` package to poll the resources for status.
 - Remove the `wait` package.

I can split this up into multiple PRs if necessary.

@monopole @seans3 @pwittrock @knverey

### CODE REVIEW

File: pkg/apply/applier.go [hunk 1]
```diff
@@ -207,7 +209,12 @@ func (a *Applier) Run(ctx context.Context) <-chan event.Event {
 		}
 
 		if a.StatusOptions.wait {
-			statusChannel := a.resolver.WaitForStatusOfObjects(ctx, infosToObjects(infos))
+			statusChannel := a.statusPoller.Poll(ctx, infosToObjMetas(infos), polling.Options{
+				PollUntilCancelled: true,
+				PollInterval:       a.StatusOptions.period,
+				UseCache:           true,
+				DesiredStatus:      status.CurrentStatus,
+			}) //.WaitForStatusOfObjects(ctx, infosToObjects(infos))
```
Comment (seans3) [inline pkg/apply/applier.go]: Should this comment be removed // .WaitForStatusOfObjects ...

File: pkg/apply/prune/inventory.go [hunk 2]
```diff
@@ -93,7 +95,7 @@ func (is *Inventory) Subtract(other *Inventory) (*Inventory, error) {
 	return result, nil
 }
 
-// Equals returns true if the "other" inventory set is the same
+// EqualsWithNormalize returns true if the "other" inventory set is the same
```
Comment (seans3) [inline pkg/apply/prune/inventory.go]: This comment is different than the name of the function.

File: pkg/apply/resource_infos.go [hunk 3]
```diff
@@ -70,7 +70,7 @@ func getIndexByKind(kind string) int {
 	return m[kind]
 }
 
-// Equals returns true if the GVK's have equal fields.
+// EqualsWithNormalize returns true if the GVK's have equal fields.
```
Comment (seans3) [inline pkg/apply/resource_infos.go]: Hmmm...doesn't match the comment does not match the function signature.

File: pkg/kstatus/polling/statusreaders/common.go [hunk 4]
```diff
@@ -137,8 +137,8 @@ func toSelector(resource *unstructured.Unstructured, path ...string) (labels.Sel
 	return metav1.LabelSelectorAsSelector(&s)
 }
 
-func toIdentifier(u *unstructured.Unstructured) wait.ResourceIdentifier {
-	return wait.ResourceIdentifier{
+func toIdentifier(u *unstructured.Unstructured) object.ObjMetadata {
```
Comment (seans3) [inline pkg/kstatus/polling/statusreaders/common.go]: Should this function go into the new object package?

File: pkg/kstatus/polling/statusreaders/common.go [hunk 5]
```diff
@@ -161,7 +161,7 @@ func getNamespaceForNamespacedResource(object runtime.Object) string {
 
 // keyForNamespacedResource returns the object key for the given identifier. It makes
 // sure to set the namespace to default if it is not provided.
-func keyForNamespacedResource(identifier wait.ResourceIdentifier) types.NamespacedName {
+func keyForNamespacedResource(identifier object.ObjMetadata) types.NamespacedName {
```
Comment (seans3) [inline pkg/kstatus/polling/statusreaders/common.go]: Should this function go in the new object package?

**Comment by seans3** [comment]
This looks good. I mention a couple nits. Thanks for pulling the ObjMetadata into a common package; very helpful. I think we should work on creating smaller PR's. As you mention this could/should be multiple PR's. I'll wait for a bit before giving the LGTM to give the others a chance for feedback.

File: pkg/apply/applier.go (hunk 1 above)
Comment (mortent) [reply pkg/apply/applier.go]: Yeah, I have removed it.

File: pkg/apply/prune/inventory.go (hunk 2 above)
Comment (mortent) [reply pkg/apply/prune/inventory.go]: This is the goland refactoring command that has been too eager. I have updated it.

File: pkg/apply/resource_infos.go (hunk 3 above)
Comment (mortent) [reply pkg/apply/resource_infos.go]: Fixed

File: pkg/kstatus/polling/statusreaders/common.go (hunk 4 above)
Comment (mortent) [reply pkg/kstatus/polling/statusreaders/common.go]: Yeah, I agree maybe we should move it. If you don't mind, I'll look at that in a separate PR.

File: pkg/kstatus/polling/statusreaders/common.go (hunk 5 above)
Comment (mortent) [reply pkg/kstatus/polling/statusreaders/common.go]: Will look at this in a separate PR.

**Pull request merged**